services/management/dbaas/components_service.go

```diff
@@ -244,37 +238,53 @@ func (c componentsService) CheckForOperatorUpdate(ctx context.Context, req *dbaa
 		close(responseCh)
 	}()
 
-	// Meanwhile, get latest operators version compatible with installed PMM.
 	pmmVersion, err := goversion.NewVersion(pmmversion.PMMVersion)
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
-	latestPXCOperatorForInstalledPMM, latestPSMDBOperatorForInstalledPMM, err := c.versionServiceClient.GetLatestOperatorVersion(ctx, pmmVersion.Core().String())
+	resp := &dbaasv1beta1.CheckForOperatorUpdateResponse{
+		ClusterToComponents: make(map[string]*dbaasv1beta1.ComponentsUpdateInformation),
+	}
+
+	latestPXCOperatorVersion, latestPSMDBOperatorVersion, err := c.versionServiceClient.GetLatestOperatorVersion(ctx, pmmVersion.Core().String())
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	resp := &dbaasv1beta1.CheckForOperatorUpdateResponse{
-		ClusterToComponents: make(map[string]*dbaasv1beta1.ComponentsUpdateInformation),
-	}
 	// Some of the requests to kuberenetes clusters for getting operators versions should be done.
 	// Go through them and decide what operator needs update.
 	for operatorsVersion := range responseCh {
-		installedPXCOperatorVersion, pxcErr := goversion.NewVersion(operatorsVersion.pxcOperatorVersion)
-		installedPSMDBOperatorVersion, psmdbErr := goversion.NewVersion(operatorsVersion.psmdbOperatorVersion)
+		// Get next operators version, don't take compatibility into account, we need to go through all versions.
```

Comment:
Do we have ticket to upgrade operators to the latest version by one click?

Reply:
Not yet